example/screenshot.dart

```diff
@@ -2,60 +2,72 @@
-import 'dart:io';
-import 'package:ffi/ffi.dart';
 
+import 'package:ffi/ffi.dart';
 import 'package:win32/win32.dart';
 
-final hInstance = GetModuleHandle(nullptr);
-
-void captureImage(int hwnd) {
-  final hdcScreen = GetDC(NULL);
-  final hdcWindow = GetDC(hwnd);
+void main() {
+  for (final display in Displays.all()) {
+    if (display.isTurnOn) {
+      display.saveScreenshot("${display.name}.bmp");
+    }
+  }
+}
 
-  final hdcMemDC = CreateCompatibleDC(hdcWindow);
-  final bmpScreen = calloc<BITMAP>();
+class Displays {
+  static Iterable<Display> all() sync* {
+    final device = calloc<DISPLAY_DEVICE>()..ref.cb = sizeOf<DISPLAY_DEVICE>();
+    var deviceIndex = 0;
 
-  try {
-    if (hdcMemDC == 0) {
-      MessageBox(
-          hwnd, TEXT('CreateCompatibleDC failed.'), TEXT('Failed'), MB_OK);
-      return;
+    while (EnumDisplayDevices(nullptr, deviceIndex, device.cast(), 0) != 0) {
+      final isTurnOn = device.ref.StateFlags != 0;
+      yield Display(device.ref.DeviceName, isTurnOn);
+      deviceIndex++;
     }
+  }
+}
 
-    final rcClient = calloc<RECT>();
-    GetClientRect(hwnd, rcClient);
+class Display {
+  final String rawName;
+  final bool isTurnOn;
 
-    SetStretchBltMode(hdcWindow, HALFTONE);
+  Display(this.rawName, this.isTurnOn);
+
+  void saveScreenshot(String fileName) {
+    final hdcScreen = _createDC();
+    try {
+      final hbmScreen = _createScreenshot(hdcScreen);
+      _saveToFile(fileName, hdcScreen, hbmScreen);
+    } finally {
+      ReleaseDC(NULL, hdcScreen);
+    }
+  }
 
-    StretchBlt(
-        hdcWindow,
-        0,
-        0,
-        rcClient.ref.right,
-        rcClient.ref.bottom,
-        hdcScreen,
-        0,
-        0,
-        GetSystemMetrics(SM_CXSCREEN),
-        GetSystemMetrics(SM_CYSCREEN),
-        SRCCOPY);
+  int _createDC() {
+    return CreateDC(nullptr, rawName.toNativeUtf16(), nullptr, nullptr);
+  }
 
-    final hbmScreen = CreateCompatibleBitmap(
-        hdcWindow,
-        rcClient.ref.right - rcClient.ref.left,
-        rcClient.ref.bottom - rcClient.ref.top);
+  int _createScreenshot(int hdcScreen) {
+    final hdcMemDC = CreateCompatibleDC(hdcScreen);
 
-    SelectObject(hdcMemDC, hbmScreen);
+    final width = GetDeviceCaps(hdcScreen, HORZRES);
+    final height = GetDeviceCaps(hdcScreen, VERTRES);
 
-    BitBlt(hdcMemDC, 0, 0, rcClient.ref.right - rcClient.ref.left,
-        rcClient.ref.bottom - rcClient.ref.top, hdcWindow, 0, 0, SRCCOPY);
+    try {
+      final hbmScreen = CreateCompatibleBitmap(hdcScreen, width, height);
+      SelectObject(hdcMemDC, hbmScreen);
+      BitBlt(hdcMemDC, 0, 0, width, height, hdcScreen, 0, 0, SRCCOPY);
+      return hbmScreen;
+    } finally {
+      DeleteObject(hdcMemDC);
+    }
+  }
 
+  void _saveToFile(String fileName, int hdcScreen, int hbmScreen) {
+    final bmpScreen = calloc<BITMAP>();
     GetObject(hbmScreen, sizeOf<BITMAP>(), bmpScreen);
 
     final bitmapFileHeader = calloc<BITMAPFILEHEADER>();
```

Comment:
Consider using an `Arena` to ensure that all these allocations are freed. In this example, the process exits quickly and it's not a problem, but others may copy this code into longer-running programs, so it's good to show best practices.